storage/txcache/concurrentMap.go

```diff
@@ -0,0 +1,134 @@
+package txcache
+
+import (
+	"sync"
+)
+
+// ConcurrentMap is a "thread" safe map of type string:Anything.
+// To avoid lock bottlenecks this map is dived to several (ShardCount) map shards.
+type ConcurrentMap struct {
+	shardCount int
+	shards     []*ConcurrentMapShard
+}
+
+// ConcurrentMapShard is a "thread" safe string to anything map.
+type ConcurrentMapShard struct {
+	maxSize      int
+	items        map[string]interface{}
+	sync.RWMutex // Read Write mutex, guards access to internal map.
+}
+
+// NewConcurrentMap creates a new concurrent map.
+func NewConcurrentMap(maxSize int, shardCount int) *ConcurrentMap {
```

Comment:
do we really need a new concurrent map ? can't we reuse the existing setup ?

Reply:
https://github.com/ElrondNetwork/concurrent-map/blob/master/concurrent_map.go also implements eviction, and we wanted to explicitly trigger eviction (and have full control over it).